pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -123,6 +123,11 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 		}
 		groupId, namespaceUID, namespace := r[0], r[1], r[2]
 
+		_, ok := namespaceMap[namespaceUID]
+		if !ok {
+			continue
```

Comment:
Shall we log an error, like we do with the ruler?
```
srv.log.Error("namespace not visible to the user", "user", c.SignedInUser.UserId, "namespace", r.NamespaceUID, "rule", r.UID)
```
https://github.com/grafana/grafana/blob/v8.2.2/pkg/services/ngalert/api/api_ruler.go#L185